messaginginapp/src/test/java/io/customer/messaginginapp/InAppMessagingStoreTest.kt

```diff
@@ -0,0 +1,372 @@
+import io.customer.messaginginapp.state.InAppMessagingAction
+import io.customer.messaginginapp.state.InAppMessagingManager
+import io.customer.messaginginapp.state.MessageState
+import io.customer.messaginginapp.testutils.core.IntegrationTest
+import io.customer.messaginginapp.type.InAppEventListener
+import io.customer.messaginginapp.type.InAppMessage
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.data.model.Region
+import io.mockk.Called
+import io.mockk.confirmVerified
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeInstanceOf
+import org.amshove.kluent.shouldContain
+import org.amshove.kluent.shouldContainAll
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class InAppMessagingStoreTest : IntegrationTest() {
+
+    private var inAppEventListener = mockk<InAppEventListener>(relaxed = true)
+
+    private val module = ModuleMessagingInApp(
+        config = MessagingInAppModuleConfig.Builder(
+            siteId = TestConstants.Keys.SITE_ID,
+            region = Region.US
+        ).setEventListener(inAppEventListener).build()
+    ).attachToSDKComponent()
+
+    private lateinit var manager: InAppMessagingManager
+
+    override fun setup(testConfig: TestConfig) {
+        super.setup(
+            testConfigurationDefault {
+                diGraph {
+                    sdk {
+                        overrideDependency<ModuleMessagingInApp>(module)
+                    }
+                }
+            }
+        )
+
+        manager = SDKComponent.inAppMessagingManager
+    }
+
+    // Helper function to set up the initial state for tests
+    private fun initializeAndSetUser() {
+        manager.dispatch(InAppMessagingAction.Initialize(siteId = String.random, dataCenter = String.random, environment = GistEnvironment.PROD))
+        manager.dispatch(InAppMessagingAction.SetUserIdentifier(String.random))
+    }
+
+    override fun teardown() {
+        manager.dispatch(InAppMessagingAction.Reset)
+        super.teardown()
+    }
+
+    @Test
+    fun givenMessagesWithDifferentPriorities_whenProcessingMessages_thenHighestPriorityMessageIsProcessedFirst() = runTest {
+        val messages = listOf(
+            Message(queueId = "1", priority = 2),
+            Message(queueId = "2", priority = 1),
+            Message(queueId = "3", priority = 3)
+        )
+
+        initializeAndSetUser()
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(messages))
+
+        val state = manager.getCurrentState()
+        state.messagesInQueue.size shouldBeEqualTo 3
+        (state.currentMessageState as? MessageState.Processing)?.message?.queueId shouldBeEqualTo "2"
```

Comment:
We should use `shouldBeInstanceOf` here instead

```suggestion
        state.currentMessageState
            .shouldBeInstanceOf<MessageState.Processing>()
            .message
            .queueId shouldBeEqualTo "2"
```